File: frontend/src/types/ris.ts
export interface OrderOut {
  id: string
  patient_id: string
  modality: string
  study_uid: string
  study_description: string | null
  referring_physician: string | null
  status: string
  priority: string
  scheduled_for: string | null
  started_at: string | null
  completed_at: string | null
  created_at: string
  updated_at: string
}

export interface StudyListItem {
  orthanc_id: string
  study_uid: string
  study_date: string | null
  study_time: string | null
  study_description: string | null
  modality: string | null
  patient_id_dicom: string | null
  patient_name_dicom: string | null
  patient_birth_date: string | null
  accession_number: string | null
  is_stable: boolean
  unlinked: boolean
  ris_order_id: string | null
  ris_order_status: string | null
  ris_study_description: string | null
  patient: { id: string; full_name: string; birth_date: string | null } | null
}

export interface StudyOut {
  id: number
  order_id: string
  orthanc_id: string
  series_count: number
  instance_count: number
  is_uploaded: boolean
  uploaded_at: string | null
}

export interface ProtocolOut {
  id: number
  order_id: string
  body: string
  impression: string | null
  is_draft: boolean
  signed_at: string | null
  signed_by: string | null
  created_at: string
  updated_at: string
}

export interface PatientStudy {
  order_id: string
  order_status: string
  study_uid: string
  modality: string | null
  created_at: string
  orthanc_id: string | null
  is_uploaded: boolean
  preview_url: string | null
  study_description?: string | null
  study_date?: string | null
  description?: string | null
  ris_order_status?: string | null
}

export interface ModalityOut {
  code: string
  name: string
  description: string | null
}

File: backend/db/schemas/ris.py
"""DTO RIS: заказы, исследования, протоколы."""

from __future__ import annotations

import uuid
from datetime import datetime

from pydantic import BaseModel, ConfigDict, Field


class OrderCreateRequest(BaseModel):
    patient_id: uuid.UUID
    modality: str = Field(min_length=2, max_length=8, description="CT / MR / DX / US / …")
    study_description: str | None = None
    referring_physician: str | None = None
    priority: str = Field(default="normal", pattern=r"^(normal|urgent|stat)$")
    scheduled_for: datetime | None = None


class OrderOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: str
    patient_id: uuid.UUID
    modality: str
    study_uid: str
    study_description: str | None
    referring_physician: str | None
    status: str
    priority: str
    scheduled_for: datetime | None
    started_at: datetime | None
    completed_at: datetime | None
    created_at: datetime
    updated_at: datetime


class OrderStatusUpdate(BaseModel):
    status: str = Field(pattern=r"^(scheduled|in_progress|completed|cancelled)$")
    note: str | None = None


class StudyOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    order_id: str
    orthanc_id: str
    series_count: int
    instance_count: int
    is_uploaded: bool
    uploaded_at: datetime | None


class ProtocolUpdate(BaseModel):
    body: str = Field(default="", max_length=65536)
    impression: str | None = Field(default=None, max_length=4096)


class ProtocolOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    order_id: str
    body: str
    impression: str | None
    is_draft: bool
    signed_at: datetime | None
    signed_by: uuid.UUID | None
    created_at: datetime
    updated_at: datetime


class ModalityOut(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    code: str
    name: str
    description: str | None
